packages/datadog-plugin-rhea/src/consumer.js

```diff
@@ -31,7 +31,12 @@ class RheaConsumerPlugin extends ConsumerPlugin {
       }
     })
 
-    if (this.config.dsmEnabled && msgObj.message) {
+    if (
+      this.config.dsmEnabled &&
+      msgObj.message &&
+      msgObj.message.delivery_annotations &&
+      msgObj.message.delivery_annotations[CONTEXT_PROPAGATION_KEY]
```

Comment:
```suggestion
      msgObj.message?.delivery_annotations?.[CONTEXT_PROPAGATION_KEY]
```
Likewise 🥳 .
In `amqplib` you guard against `message` being undefined, but here you don't. Worth aligning on one kind of check?